pkgs/build-support/libredirect/default.nix

```diff
@@ -45,16 +53,15 @@ else stdenv.mkDerivation rec {
       -isystem ${llvmPackages_13.libclang.lib}/lib/clang/*/include \
       -L${llvmPackages_13.clang.libc}/lib \
       -Wl,-install_name,$libName \
-      -Wall -std=c99 -O3 -fPIC libredirect.c \
-      -shared -o "$libName"
     '' else if stdenv.isDarwin then ''
-    $CC -Wall -std=c99 -O3 -fPIC libredirect.c \
+    $CC \
       -Wl,-install_name,$out/lib/$libName \
-      -shared -o "$libName"
     '' else ''
-    $CC -Wall -std=c99 -O3 -fPIC libredirect.c \
+    $CC \
+    ''} \
+      -Wall -std=c99 -O3 -fPIC libredirect.c \
+      -DENABLE_TEST_SYSTEM=$ENABLE_TEST_SYSTEM \
       -shared -o "$libName"
-    ''}
```

Comment:
Did another commit get mixed in here somehow?  I don't understand why this is a ~8 line change instead of a ~1 line change, but maybe I'm missing something.

Reply:
This includes a small refactoring.